src/core/morphing.js

```diff
@@ -0,0 +1,110 @@
+import { Idiomorph } from "idiomorph/dist/idiomorph.esm.js"
+import { FrameElement } from "../elements/frame_element"
+import { dispatch } from "../util"
+
+export function morphRefresh(currentElement, newElement) {
```

Comment:
What do you think about moving the logic of `morphRefresh` to `PageMorphRenderer` and `morphFrames` to `FrameMorphRenderer`. We can leave this `morphing.js` module in charge of lower-level primitives, but I think it's fine to move this higher-level behavior to those rendererer classes. For example, `canRefreshFrame`, could be just a private method in `PageMorphRenderer`.

Reply:
I'm happy to move those functions in this commit. 

A secondary goal for this refactoring was in preparation for public consumption of morphing with something like `event.detail.render = Turbo.morphElements(...)` or `event.detail.render = Turbo.morphFrames(...)` in `turbo:before-render` and `turbo:before-frame-render` event listeners.

Exposing these methods publicly, then consuming them internally would provide consistent morphing (and event dispatching) throughout the codebase. Defining those methods on the renderer would force applications interested in experimenting with morphing to reach through the internal package structure to access the renderers.

Depending on our goals here, enabling access to internals in the absence of a public API might be the best way forward for now, until a design decisions for morphing (or [Frame-specific refreshes](https://github.com/hotwired/turbo/issues/1161), or [morphing across URLs with differing pathnames](https://github.com/hotwired/turbo/issues/1177), etc.) are settled.

Reply:
@seanpdoyle if we moved `morphPage` and `morphFrames` to the renderer objects, couldn't you then invoke `MorphingFrameRenderer.renderElement` and `MorphingPageRenderer.renderElement` if you just wanted to reuse the high level morphing+event dispatching logic from other places? 

The problem I see with the current API `morphFrames`, `morphPage`, `morphElements`, `morphChildren` is that it mixes levels of abstraction. It lacks a bit of clarity. Leveraging the renderer classes to encapsulate the operations at the higher level of abstraction, and leaving the module for the low-level primitives make sense to me.

Reply:
@jorgemanrubia I've re-structured that module to only export `morphElements` and `morphChildren`. 

In the interest of merging this re-structuring work, I've done my best to make sure these changes don't affect the public API.

If this work is merged, I'm planning on taking a few next steps:

1. exporting `Turbo.morphPage` as an alias for `MorphingPageRenderer.renderElement`
2. exporting `Turbo.morphFrame` as an alias for `MorphingFrameRenderer.renderElement`
3. exporting `Turbo.morphElements` and `Turbo.morphChildren` as-is
4. exploring ways to make `FrameElement.reload` integrate with morphing without needing to define a `turbo:before-frame-render` event listener

As a follow-up to this PR, reviewing and merging https://github.com/hotwired/turbo/pull/1028 andhttps://github.com/hotwired/turbo/pull/1027 might help reduce the number of moving parts.

At the moment, the `MorphingFrameRenderer` isn't utilized anywhere, and only serves as a class to hook a `static renderElement` function off of. In order to achieve 4. from the list above, there might be an opportunity to re-arrange how Frame rendering is determined. I've opened https://github.com/hotwired/turbo/pull/430 to explore what it'd take to introduce a `FrameVisit` corollary to a `Visit`, and plan on catching that branch up with these newer changes.